index.js

```diff
@@ -1,15 +1,65 @@
-const port = parseInt(process.env.PORT, 10) || 3000;
 const app = require("express")();
 const bodyParser = require("body-parser");
 const xrefResponseBody = require("respec-xref-route");
-const xrefData = require("./xref-data.json");
+const crypto = require("crypto");
+const { exec } = require("child_process");
 
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(bodyParser.json());
+let xrefData = require("./xref-data.json");
+
+const conf = {
+  port: parseInt(process.env.PORT, 10) || 3000,
+  bikeshedSecret: process.env.BIKESHED_SECRET || "",
+};
+
+app.use(bodyParser.urlencoded({ verify: rawBodyParser, extended: false }));
+app.use(bodyParser.json({ verify: rawBodyParser }));
 
 app.post("/xref", (req, res) => {
   const body = xrefResponseBody(req.body, xrefData);
   res.json(body);
 });
 
-app.listen(port, () => console.log(`Listening on port ${port}!`));
+app.post("/xref/update", handleBikeshedUpdate);
```

Comment:
This should be something like:

```
app.post("/xref/update", require("./routes/xref-update-route").route);
```

Reply:
Why is there a `.route` in end? 

Also, how about the middlewares? I think might be a good idea to export an array from each route as:
``` js
module.exports = [bodyParser({}), validateGithubSignature, updateData];
```

and use as 
``` js
app.post("/xref/update", ...require("./routes/xref-update-route"));
```